datavec-jdbc/src/main/java/org/datavec/api/records/reader/impl/jdbc/JDBCRecordReader.java

```
@@ -0,0 +1,145 @@
+            statement = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+            statement.closeOnCompletion();
+            ResultSet rs = statement.executeQuery(this.query);
+            this.meta = rs.getMetaData();
+            this.iter = new ResettableResultSetIterator(rs);
+        } catch (SQLException e) {
+            closeJdbc();
+            throw new RuntimeException("Could not connect to the database", e);
+        }
+    }
+
+    @Override
+    public void initialize(Configuration conf, InputSplit split) throws IOException, InterruptedException {
+
+    }
+
+    @Override
+    public List<Writable> next() {
+        List<Writable> ret = new ArrayList<>();
+        if (iter.hasNext()) {
+            Object[] next = iter.next();
+            invokeListeners(next);
+            for (int i = 0; i < next.length; i++) {
+                try {
+                    Object columnValue = next[i];
+                    if (trimStrings && columnValue instanceof String) {
+                        columnValue = ((String) columnValue).trim();
+                    }
+                    // Note, getColumnType first argument is column number starting from 1
+                    Writable writable = JdbcWritableConverter.convert(columnValue, meta.getColumnType(i + 1));
+                    ret.add(writable);
+                } catch (SQLException e) {
+                    closeJdbc();
+                    throw new RuntimeException("Error reading database metadata");
+                }
+            }
+        }
+
+        return ret;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return iter.hasNext();
+    }
+
+    @Override
+    public List<String> getLabels() {
+        return null;
+    }
+
+    @Override
+    public void reset() {
+        iter.reset();
+    }
+
+    @Override
+    public List<Writable> record(URI uri, DataInputStream dataInputStream) throws IOException {
```

throw an UnsupportedOperationExecption instead of returning null